src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -239,6 +240,20 @@ module.exports = function<T, P, I, TI, C, CX>(
             currentHostContext,
             workInProgress
           );
+
+          // A component cannot be focused until it's mounted.
+          // Flag components with autoFocus for later work.
+          switch (type) {
+            case 'button':
+            case 'input':
+            case 'select':
+            case 'textarea':
+              if (newProps.autoFocus) {
```

Comment:
Right 👍 Sorry. This wasn't quite ready for review yet. I probably shouldn't have opened a PR yet. I just like to track my thoughts-in-progress via PR.